graphql_compiler/compiler/emit_sql.py

```diff
@@ -126,6 +134,94 @@ def _find_folded_fields(ir):
     return folded_fields
 
 
+class XMLPathBinaryExpression(BinaryExpression):
+    """Special override of BinaryExpression used to trigger `compile_xmlpath` during compile.
+
+    This type of binary expression is used to describe the Selectable selected inside the
+    XML PATH subquery. Using an XMLPathBinaryExpression forces the compiler to produce that
+    Selectable without aliasing (aka labeling) it. This prevents the string resulting from
+    the XML PATH from having extraneous XML tags based on the alias given to the column.
+    """
+
+
+@compiles(_CompileLabel, 'default')
+def compile_xmlpath(element, compiler, **kw):
+    """Suppress labeling when compiling XML PATH subqueries, otherwise compile as usual.
+
+    This method is needed because if XML PATH column is selected with an alias then each
+    entry in the resulting array is wrapped in XML tags bearing that alias.
+    e.g. `SELECT (SELECT ... FOR XML PATH ('')) AS blah ...` results in a string of the form
+    <blah>element1</blah><blah>element2</blah>. Since SQLAlchemy adds labels to all Selectables, we
+    must to suppress that labeling.
+    """
+    if isinstance(element.element, XMLPathBinaryExpression):
+        # this indicates that no label should be inserted for this element
+        kw.update(
+            within_columns_clause=False  # label always gets added if within_columns_clause is True
+        )
+    return compiler.visit_label(element, **kw)
+
+
+def get_xml_path_clause(output_column, where):
+    """Produce an MSSQL-style XML PATH-based aggregation subquery.
+
+    XML PATH clause aggregates the values of the output_column from the output vertex
+    properly encoded to ensure that we can reconstruct a list of the original values
+    regardless of what characters they contain, whether they are empty or null, and
+    regardless of their data type.
+
+    Empty arrays are represented as '', null values are represented as '~', array
+    entries are delimited using '|'.
+
+    All occurrences of '^', '~', and '|' in the original string values are
+    replaced with '^e' (escape), '^n' (null), and '^d' (delimiter), resp.
+
+    Undoing the encoding above as well as the XML reference entity encoding performed
+    by the XML PATH statement is deferred to post-processing when the list is retrieved
+    from the string representation produced by the subquery.
+
+    Post-processing must split on '|', convert '~' to None, and undo both the encoding above
+    and the XML reference entity encoding auto performed by XML PATH. In particular, undo
+    '^d' -> '|', '^e' -> '^', '^n' -> '~', '&amp;' -> '&', '&gt;' -> '>', '&lt;' -> '<',
+    '&0xHEX;' -> u'0xHEX'. XML post processing can be done with
```

Comment:
For the `&0xHEX` cases, do you know what characters trigger that case and what the hex value is referring to? I suspect there's an issue with the suggested `u'0xHEX'` transformation since I think hex codes don't have any meaning in Unicode since Unicode deals in code points, not bytes.

Separately, it might be good to include an example, perhaps in parens, of a specific character that got escaped into `&0xHEX` form and what it should be unescaped into. This would help implementers of this post-processing verify the correctness of their algorithm by providing a ready-made test case.

Reply:
Let me know if this explanation is what you had in mind